client/my-sites/stats/stats-notices/index.tsx

```diff
@@ -22,6 +39,28 @@ const NewStatsNotices = ( { siteId, isOdysseyStats }: StatsNoticesProps ) => {
 		isJetpackSite( state, siteId, { treatAtomicAsJetpackSite: false } )
 	);
 
+	const reduxDispatch = useDispatch();
+	// TODO: Display error messages on the notice.
+	// eslint-disable-next-line @typescript-eslint/no-unused-vars
+	const [ error, setError ] = useState< Error | null >( null );
+
+	// Update site purchases in Redux store for Odyssey Stats.
+	useEffect( () => {
+		if ( isOdysseyStats ) {
+			wpcom.req
+				.get( { path: '/site/purchases', apiNamespace: 'jetpack/v4' } )
+				.then( ( res: { data: string } ) => JSON.parse( res.data ) )
+				.then( ( purchases: Record< string, string >[] ) => {
+					reduxDispatch( {
+						type: SITE_PURCHASES_UPDATE,
+						siteId,
+						purchases: formatPurchasesToSiteProducts( purchases ),
+					} );
+				} )
+				.catch( ( error: Error ) => setError( error ) );
+		}
```

Comment:
Do we anticipate needing this purchase data in multiple components within Odyssey in the near future?

Reply:
not shared too much but it's already a state in Calypso, and we want it to be up to date all the time if that makes sense.